src/security/Handshake.h

```diff
@@ -120,6 +131,42 @@ class HandshakeParser
     YesNoNone expectingModernRecords;
 };
 
+/// \returns true when the given protocol is TLS v1.3 or later
+/// Negating the result of this function may be misleading. For example,
+/// !Tls1p3orLater(x) does not imply that x is TLS v1.2 or lower; x could also
+/// be unknown or even HTTP/2. Use Tls1p2orEarlier() instead of negating.
+inline bool
+Tls1p3orLater(const AnyP::ProtocolVersion &version)
+{
+    return version.protocol == AnyP::PROTO_TLS &&
+           version >= AnyP::ProtocolVersion(AnyP::PROTO_TLS, 1, 3);
+}
+
+/// \returns true when the given protocol is TLS v1.2 or earlier, including SSL
+/// Negating the result of this function may be misleading. See Tls1p3orLater()
+/// for the discussion and a better alternative to negation.
+inline bool
+Tls1p2orEarlier(const AnyP::ProtocolVersion &version)
+{
+    return version.protocol == AnyP::PROTO_SSL ||
+           (version.protocol == AnyP::PROTO_TLS &&
+            version <= AnyP::ProtocolVersion(AnyP::PROTO_TLS, 1, 2));
```

Comment:
The TlsVersionEarlierThan function compares  two TLS/SSL versions it make sense to add the Must clause to be sure that the two versions are refers to TLS/SSL protocols.
Currently it is used inside parseSupportedVersionsExtension and the Musts will never throw. However in the future maybe will be used in a place where the check is required.

The Tls1p3orLater and Tls1p2orEarlier functions checks one protocol. If the protocol is not an SSL/TLS then it is not TLSv1.3 or later (or TLSv1.2 or earlier). Why do we throw an assertion instead of just return a false?

Reply:
> Why do we throw an assertion instead of just return a false?

... because the false result is very easy to misinterpret and/or because the comparison function is not defined (i.e. has no reasonable meaning) for other protocols.

Reply:
I have not strong opinion on this. The results can be misinterpreted however the Tls1p3orLater/Tls1p2orEarlier functions are not lied when return a false on an non TLS/SSL protocol.
The decision is yours :)

Reply:
If you do not see a problem with the suggested change, please implement it.

Reply:
commit 99f673b8

Reply:
> commit 99f673b8

Please do not forget to push that commit. It is not on GitHub right now.

Reply:
> Please do not forget to push that commit. It is not on GitHub right now.

Sorry, It should be ok now.